### DISCUSSION TITLE
Reconcilation of Teleport logs (and session recordings) with Application logs

### DISCUSSION THREAD

**Discussion by asundeupgrade** [gravitational/teleport#22238] [Q&A]
Hello,

I'm trying to find a way to link my application logs with a teleport session. 

The scenario is that a user through teleport will curl an endpoint. The application behind the endpoint will produce some audit logs of it's own. 

`tsh ssh -L 5000:myEndpoint:80 --local node curl http://localhost:5000`

I'm looking for a way to connect the teleport logs with the application's audit logs via perhaps a sessionId? Is there anything injected into the curl by teleport's proxy that I could use on the application side? Or perhaps an API the application can call to get the session information?

Thank you

**Comment by webvictim** [comment]
As you're using SSH port forwarding to facilitate the connection, there will be no logs inside Teleport of the traffic to/from the application. You'll see an SSH port forwarding event and nothing more.

However, if you were to change your configuration to use Teleport application access instead (as per https://goteleport.com/docs/application-access/guides/) then you'd be able to get full web session logs for every user, similar to those described here: https://github.com/gravitational/teleport/discussions/11334#discussioncomment-2415029

**Reply by asundeupgrade** [reply]
Thanks, yes that is an improvement on the web session, but at this point I'm more interested in the downstream side effects that you cannot see in the web session. I'm looking for a way to connect the teleport session to what is happening internally within the application.

**Reply by webvictim** [reply]
I see what you're saying. One cool way we could do this would be to inject the Teleport application session ID as a header into every request that Teleport makes to the application. I've raised a [feature request here](https://github.com/gravitational/teleport/issues/22417) - please feel free to comment/+1 if that'd work for you.

Unfortunately I don't know of an easy way to do this currently.
